File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        // recursive relation: dp[n] = dp[n-1] + dp[n-2]
        if (n == 1) return 1;
        if (n == 2) return 2;

        // 1, 2, 3, 5, 8, 11
        int first = 1, second = 2;
        for (int i = 3; i <= n; i++) {
            int temp = first + second;
            first = second;
            second = temp;
        }

        return second;
    }
};

File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        
        if n == 2:
            return 2
        
        prev, curr = 1, 2
        for i in range(3, n+1):
            temp = curr + prev
            prev = curr
            curr = temp
        
        return curr